controllers/capabilities/developeraccount_status_reconciler.go

```diff
@@ -0,0 +1,171 @@
+	return &DeveloperAccountStatusReconciler{
+		BaseReconciler:         b,
+		resource:               resource,
+		providerAccountHost:    providerAccountHost,
+		remoteDeveloperAccount: remoteDeveloperAccount,
+		reconcileError:         reconcileError,
+		logger:                 b.Logger().WithValues("Status Reconciler", resource.Name),
+	}
+}
+
+func (s *DeveloperAccountStatusReconciler) Reconcile() (reconcile.Result, error) {
+	s.logger.V(1).Info("START")
+
+	newStatus, err := s.calculateStatus()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	equalStatus := s.resource.Status.Equals(newStatus, s.logger)
+	s.logger.V(1).Info("Status", "status is different", !equalStatus)
+	s.logger.V(1).Info("Status", "generation is different", s.resource.Generation != s.resource.Status.ObservedGeneration)
+	if equalStatus && s.resource.Generation == s.resource.Status.ObservedGeneration {
+		// Steady state
+		s.logger.V(1).Info("Status steady state, status was not updated")
+		return reconcile.Result{}, nil
+	}
+
+	// Save the generation number we acted on, otherwise we might wrongfully indicate
+	// that we've seen a spec update when we retry.
+	// TODO: This can clobber an update if we allow multiple agents to write to the
+	// same status.
+	newStatus.ObservedGeneration = s.resource.Generation
+
+	s.logger.V(1).Info("Updating Status", "sequence no:", fmt.Sprintf("sequence No: %v->%v", s.resource.Status.ObservedGeneration, newStatus.ObservedGeneration))
+
+	s.resource.Status = *newStatus
+	updateErr := s.Client().Status().Update(s.Context(), s.resource)
+	if updateErr != nil {
+		// Ignore conflicts, resource might just be outdated.
+		if errors.IsConflict(updateErr) {
+			s.logger.Info("Failed to update status: resource might just be outdated")
+			return reconcile.Result{Requeue: true}, nil
+		}
+
+		return reconcile.Result{}, fmt.Errorf("Failed to update status: %w", updateErr)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (s *DeveloperAccountStatusReconciler) calculateStatus() (*capabilitiesv1beta1.DeveloperAccountStatus, error) {
+	// Initialize with existing data for data coming from 3scale
+	// just in case in this reconciliation loop something goes wrong and avoid replacing right data with nil
+	newStatus := &capabilitiesv1beta1.DeveloperAccountStatus{
+		ID:                  s.resource.Status.ID,
+		AccountState:        s.resource.Status.AccountState,
+		CreditCardStored:    s.resource.Status.CreditCardStored,
+		ProviderAccountHost: s.resource.Status.ProviderAccountHost,
+		Conditions:          s.resource.Status.Conditions.Copy(),
+		ObservedGeneration:  s.resource.Status.ObservedGeneration,
+	}
+
+	// If there is an error and s.remoteDeveloperAccount is nil
+	// ID is not changed, as it is initialized with previous values d
```

Comment:
Is the comment incomplete?

I' trying to understand this use case

Reply:
This comment does not make sense here, the message was added few lines above. 

The use case is that `status.ID` has been set correctly. In later reconciliation loop, there is some unexpected error and `s.remoteDeveloperAccount` (the 3scale developer account object read from the API) can be nil. Therefore, the `status.ID` is not changed to nil, instead, the existing value remains untouched.